scico/typing.py.orig

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2021-2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SPORCO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+"""Type definitions."""
+
+from typing import Any, Tuple, Union
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+
+__author__ = """Luke Pfister <luke.pfister@gmail.com>"""
+
+
+JaxArray = Union[jax.interpreters.xla.DeviceArray, jax.interpreters.pxla.ShardedDeviceArray]
+"""A jax array."""
+
+Array = Union[np.ndarray, JaxArray]
+"""Either a numpy or jax array."""
+
+PRNGKey = jnp.ndarray
+"""A key for jax random number generators (see :mod:`jax.random`)."""
+
+DType = Any
+"""A numpy or jax dtype."""
+
+Shape = Tuple[int, ...]
+"""A shape of a numpy or jax array."""
+
+BlockShape = Tuple[Tuple[int, ...], ...]
+"""A shape of a :class:`.BlockArray`."""
+
+<<<<<<< HEAD
```

Comment:
leftover merge markings

Reply:
Good catch. Not sure how this file ended up in the repo. It's gone now.

Reply:
Good catch! This file was added by mistake. It's gone now.